api/invite.go

```diff
@@ -526,3 +851,357 @@ func (a *Api) SendInvite(res http.ResponseWriter, req *http.Request, vars map[st
+					"Email":                    invite.Email,
+					"WebPath":                  webPath,
+				}
+
+				if a.createAndSendNotification(req, invite, emailContent, inviteeLanguage) {
+					a.logAudit(req, "invite sent")
+				} else {
+					a.logAudit(req, "invite failed to be sent")
+					log.Print("Something happened generating an invite email")
+					res.WriteHeader(http.StatusUnprocessableEntity)
+					return
+				}
+			}
+
+			a.sendModelAsResWithStatus(res, invite, http.StatusOK)
+			return
+		}
+	}
+
+}
+
+// @Summary Send notification to an hcp that becomes admin
+// @Description  Send an email to th enew admin
+// @ID hydrophone-api-UpdateTeamInvite
+// @Accept  json
+// @Produce  json
+// @Success 200 {object} models.Confirmation "invite details"
+// @Failure 400 {object} status.Status "userId, teamId and isAdmin were not provided or the payload is missing/malformed"
+// @Failure 401 {object} status.Status "Authorization token is missing or does not provide sufficient privileges"
+// @Failure 403 {object} status.Status "Authorization token is invalid"
+// @Failure 409 {object} status.Status "No notification and email sent; User is already an admin"
+// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service"
+// @Failure 500 {object} status.Status "Internal error while processing the invite, detailled error returned in the body"// @Router /confirm/send/team-invite [put]
+// @security TidepoolAuth
+func (a *Api) UpdateTeamInvite(res http.ResponseWriter, req *http.Request, vars map[string]string) {
+	// By default, the invitee language will be "en" for Englih (as we don't know which language suits him)
+	// In case the invitee is a known user, the language will be overriden in a later step
+	var inviteeLanguage = "en"
+	tokenValue := req.Header.Get(TP_SESSION_TOKEN)
+	token := a.token(res, req)
+	if token == nil {
+		return
+	}
+
+	invitorID := token.UserID
+	if invitorID == "" {
+		res.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	defer req.Body.Close()
+	var ib = &inviteBody{}
+	if err := json.NewDecoder(req.Body).Decode(ib); err != nil {
+		log.Printf("UpdateInvite: error decoding invite to detail %v\n", err)
+		statusErr := &status.StatusError{Status: status.NewStatus(http.StatusBadRequest, STATUS_ERR_DECODING_INVITE)}
+		a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+		return
+	}
+
+	if ib.Email == "" || ib.TeamID == "" {
+		statusErr := &status.StatusError{Status: status.NewStatus(http.StatusBadRequest, STATUS_ERR_MISSING_DATA_INVITE)}
+		a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+		return
+	}
+	if ib.IsAdmin == "" {
+		ib.IsAdmin = "false"
+	}
+
+	_, team, err := a.userIsTeamAdmin(tokenValue, token.UserID, ib.TeamID, res)
+	if err != nil {
+		return
+	}
+
+	var isAdmin = strings.ToLower(ib.IsAdmin) == "true"
+	if isAdmin {
+		invitedUsr := a.findExistingUser(ib.Email, a.sl.TokenProvide())
```

Comment:
when user id is known (user already registered) can't we have this information on the payload?
We could also authorize the ChangeRole (member->admin) use case only for registered users, this would simplify the process

Reply:
that will be the case. If we cannot find the user, we return an error

Reply:
I meant, in the invitation body, so we don't need to call shoreline again.